modules/product/src/Plugin/Field/FieldWidget/ProductVariationTitleWidget.php

```diff
@@ -124,7 +126,8 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
       '#title' => $this->getSetting('label_text'),
       '#options' => $variation_options,
       '#required' => TRUE,
-      '#default_value' => $selected_variation->id(),
+      '#default_value' => $selected_variation ? $selected_variation->id() : NULL,
+      '#attributes' => ['data-disable-refocus' => 'true'],
```

Comment:
> That is, when you choose one of the options in the second, or third ,,,etc, select input, the focus jumps to the first select input.

This only fixes this issue for title widget, not attributes. Or is issue here?

Reply:
The issue of "jumping" also exists (not fixed yet) on the ProductVariationAttributesWidget. It could be fixed with rejecting refocusing using HTML attribute as the above example or explicitly assigning persistent ID for a widget:

``` 
// .....
$element['variation'] = [
  '#type' => 'select',
  '#id' => "edit-purchased-entity-0-product-{$product->id()}-variation-{$selected_variation->id()}",
  '#title' => $this->getSetting('label_text'),
  '#options' => $variation_options,
  '#required' => TRUE,
  '#default_value' => $selected_variation->id(),
  '#ajax' => [
    'callback' => [get_class($this), 'ajaxRefresh'],
    'wrapper' => $form['#wrapper_id'],
  ],
];
// ....
```

Reply:
As @drugan told this jumping can be fixed by adding unique id for variation form.
Just mixed his code from his project with this problem here and got working code.
There where issues with jumping and not selecting the variations as planned.

    $base_id = $form_state->getFormObject()->getBaseFormId();
    $id = strtr($form_state->getFormObject()->getFormId(), [$base_id => '']);
    foreach ($this->getAttributeInfo($selected_variation, $variations) as $field_name => $attribute) {
      $element['attributes'][$field_name] = [
        '#id' => 'edit-purchased-entity-0-attributes-' . Html::getClass($field_name . $id),
        '#type' => $attribute['element_type'],
        '#title' => $attribute['title'],
        '#options' => $attribute['values'],
        '#required' => $attribute['required'],
        '#default_value' => $selected_variation->getAttributeValueId($field_name),
        '#limit_validation_errors' => [],
        '#ajax' => [
          'callback' => [get_class($this), 'ajaxRefresh'],
          'wrapper' => $form['#wrapper_id'],
        ],
      ];

Reply:
[fix_jumping.patch.txt](https://github.com/drupalcommerce/commerce/files/1775935/fix_jumping.patch.txt)